### DISCUSSION TITLE
Frequent occurrence X-Vercel-Error: ROUTER_EXTERNAL_TARGET_CONNECTION_ERROR

### DISCUSSION THREAD

**Discussion by xyfacai** [vercel/community#2838] [Help] [answered]
### Summary

I will deploy my website on Vercel for testing.
Packaging and deployment are normal, and the website appears to be running normally
But sometimes when fetching my API, it will make normal requests, and sometimes it will be abnormal, returning this error
X-Vercel-Error: ROUTER_EXTERNAL_TARGET_CONNECTION_ERROR

He has really troubled me for a long time, I don't understand why. I am an Hobby user.

### Steps to Reproduce

X-Vercel-Error: ROUTER_EXTERNAL_TARGET_CONNECTION_ERROR
![image](https://github.com/vercel/community/assets/52985669/241ef1b7-5555-448c-abc5-066176ceb2cb)
![image](https://github.com/vercel/community/assets/52985669/49900342-d382-4b89-a217-a761a20a234b)

It can be confirmed that this request did not request my backend service

This is my fetch code
```
  fetch() {
    if (fetchState > 0) return Promise.resolve();

    fetchState = 1;

    return fetch("backend/v1/user/account/info", {
      method: "get",
      body: null,
      headers: {
        ...getHeaders(),
      },
      credentials: "include",
    })
  },
```

This is my next.config.mjs
```
const nextConfig = {
  experimental: {
    appDir: true,
  },
  async rewrites() {
    const ret = [
      {
        source: "/google-fonts/:path*",
        destination: "https://fonts.googleapis.com/:path*",
      },
      {
        source: "/backend/:path*",
        destination: "http://123.456.789.789:80/backend/:path*",
      }
    ];
    
    return {
      beforeFiles: ret,
    };
  },
  webpack(config) {
    config.module.rules.push({
      test: /\.svg$/,
      use: ["@svgr/webpack"],
    });

    return config;
  },
  output: "standalone",
};

```

**Comment by smaeda-ks** [comment, marked as answer]
@xyfacai I suspect your origin is blocking our IPs. Could you check your security settings and see it's blocking any IPs?

**Reply by xyfacai** [reply]
I have checked the settings at my cloud service provider, and my firewall has already been set to allow requests from all IP origin. I just adjusted it to all IPs and all protocols, but the same issue still exists. 
All protection for my service has been removed

I cannot rule out that my cloud service provider itself has been blocked, but I have not found any blocking records. I am using Tencent Cloud

**Reply by smaeda-ks** [reply]
@xyfacai we're still seeing timeout when initiating a TCP connection with your server, which likely suggests IP blocking on their end. Could you reach out to Tencent Cloud support as well? They may have some insights there.

In fact, I can reach the same IP/port from my home network fine:

```
$ nc -vz 123.2xx.xxx.xxx 8xxx
Connection to 123.2xx.xxx.xxx port 8xxx [tcp/bctp] succeeded!
```

**Reply by xyfacai** [reply]
I have inquired with them and have not replied yet
So Vercel's IP is dynamic? 
Are there settings that allow for setting fixed IPs, or are they fixed within certain IP ranges?
Because most of the time, it's working

**Reply by smaeda-ks** [reply]
@xyfacai Dynamic IPs. Only Enterprise customers can get fixed IPs:
https://vercel.com/guides/can-i-get-a-fixed-ip-address

**Reply by xyfacai** [reply]
Thank you for your help. 
I am waiting for a response from Tencent Cloud technical support personnel.

If it is indeed a dynamic IP issue and Tencent Cloud cannot solve it, I am willing to pay for it.

As a backend developer, I really like Vercel

If there is any progress, I will come here to reply

**Reply by xyfacai** [reply]
I solved this problem by requesting the full address via fetch, which needed to setting the Access-Control-Allow-Origin
